fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexScanParameters.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * IndexScanParameters.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.EvaluationContext;
+import com.apple.foundationdb.record.IndexScanType;
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.query.plan.temp.explain.Attribute;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nonnull;
+
+/**
+ * The parameterized form of index scan bounds.
+ * These parameters are stored in the plan and then bound to the context before passing on to the index maintainer.
+ */
+@API(API.Status.MAINTAINED)
+public interface IndexScanParameters extends PlanHashable {
+    /**
+     * Get the type of index scan to be performed.
+     * @return the scan type
+     */
+    @Nonnull
+    IndexScanType getScanType();
+
+    /**
+     * Get the bound form of the index scan for use by the maintainer.
+     * @param store store against which the scan will be performed
+     * @param context query parameters for the scan
+     * @return the index scan bounds
+     */
+    @Nonnull
+    IndexScanBounds bind(@Nonnull FDBRecordStoreBase<?> store, @Nonnull EvaluationContext context);
```

Comment:
I think this method's purpose confused me a bit, but I think I understand this correctly:

1. An `IndexScanParameters` can have unbound parameters and is therefore used for planning whereas 
2. An `IndexScanBounds` should bind those parameters to concrete values so that it can be actual executed

And so this `bind` method is intended to translate one to the other by performing that evaluation. Is that the idea?

Reply:
I added another sentence explaining this.